controllers/awsmachine_controller.go

```diff
@@ -384,28 +426,44 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 
 	if !machineScope.Cluster.Status.InfrastructureReady {
 		machineScope.Info("Cluster infrastructure is not ready yet")
+		conditions.MarkFalse(machineScope.AWSMachine, infrav1.InstanceReadyCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
 		return ctrl.Result{}, nil
 	}
 
 	// Make sure bootstrap data is available and populated.
 	if machineScope.Machine.Spec.Bootstrap.DataSecretName == nil {
 		machineScope.Info("Bootstrap data secret reference is not yet available")
+		conditions.MarkFalse(machineScope.AWSMachine, infrav1.InstanceReadyCondition, infrav1.WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
 		return ctrl.Result{}, nil
 	}
 
 	ec2svc := r.getEC2Service(clusterScope)
 
-	// Get or create the instance.
-	instance, err := r.getOrCreate(machineScope, ec2svc, secretSvc)
+	// Find existing instance
+	instance, err := r.findInstance(machineScope, ec2svc)
 	if err != nil {
+		conditions.MarkUnknown(machineScope.AWSMachine, infrav1.InstanceReadyCondition, infrav1.InstanceNotFoundReason, err.Error())
 		return ctrl.Result{}, err
 	}
+	// Create new instance
+	if instance == nil {
+		conditions.MarkFalse(machineScope.AWSMachine, infrav1.InstanceReadyCondition, infrav1.InstanceProvisionStartedReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
If there are persistent problems in `createInstance` here we will have condition flickering at every reconciliation (due to the jump from `InstanceProvisionStartedReason` to `InstanceProvisionFailedReason`).
A possible approach to avoid this problem is to wrap this call into a 
```
if conditions.GetReason(machineScope.AWSMachine, infrav1.InstanceReadyCondition) != infrav1.InstanceProvisionFailedReason {
   conditions.MarkFalse(...)
}
```
And to add a comment explaining why we are adding this if